### DISCUSSION TITLE
Only include work items with a particular tag?

### DISCUSSION THREAD

**Discussion by williamayerst** [rfennell/AzurePipelines#1375] [Q&A] [answered]
Hi there,

I have a requirement to filter work items to only include those that have a specific tag, is there some way to action this? 

Many thanks,

**Comment by rfennell** [comment, marked as answer]
Yes, the basic answer is in this discussion thread #1349.

The issue you might have is that the tags, as I remember, are a comma separated list or might be an array, so a simple string match will not work, you will need to use something like [occurrences](https://github.com/helpers/handlebars-helpers#occurrences) or [contains](https://github.com/helpers/handlebars-helpers#contains) 
 or create your own [custom Handlebars extension](https://github.com/rfennell/AzurePipelines/tree/00b58866916b45dc08db3dfeda7e8d968cad8baa/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203)

Best bet is to just display the tags in you report, to see the format, then add the matching type of filter

**Reply by williamayerst** [reply]
Thank you - sorry for the confusion, clearly this is not specifically something relating to XplatGenerateReleaseNotes but may I ask how  I might address the tags field?

**Reply by rfennell** [reply]
The workitem tags are just a single field with a comma separated list, so to list them

```
{{#forEach workItems}}
*  **{{this.id}}**  {{lookup this.fields 'System.Title'}}
   - **WIT** {{lookup this.fields 'System.WorkItemType'}} 
   - **Tags** {{lookup this.fields 'System.Tags'}}
   - **Assigned** {{#with (lookup this.fields 'System.AssignedTo')}} {{displayName}} {{/with}}
{{/forEach}}
```
and your filter would be something like, not I have not tested this (the [local tester is useful here](https://github.com/rfennell/AzurePipelines/tree/main/Extensions/XplatGenerateReleaseNotes/V3/testconsole) to work out the template syntax right)

```
{{#forEach workItems}}
{{#contains (lookup this.fields 'System.Tags') "foo"}}
*  **{{this.id}}**  {{lookup this.fields 'System.Title'}}
   - **WIT** {{lookup this.fields 'System.WorkItemType'}} 
   - **Tags** {{lookup this.fields 'System.Tags'}}
   - **Assigned** {{#with (lookup this.fields 'System.AssignedTo')}} {{displayName}} {{/with}}
{{/contains}}
{/forEach}}
```
